subsys/net/lib/sockets/getaddrinfo.c

```diff
@@ -155,9 +157,26 @@ int zsock_getaddrinfo(const char *host, const char *service,
 	}
 
 	/* Mark entry as last */
-	ai_arr[ai_state.idx - 1].ai_next = NULL;
-
-	*res = ai_arr;
+	ai_state.ai_arr[ai_state.idx - 1].ai_next = NULL;
 
 	return 0;
 }
+
+
+int zsock_getaddrinfo(const char *host, const char *service,
+		      const struct zsock_addrinfo *hints,
+		      struct zsock_addrinfo **res)
```

Comment:
(Probably just being paranoid.) Do we need to check res to be non-null before calloc()?

Reply:
This function will run in user mode, so the penalty for passing either a NULL or a nonsense memory address will be an exception which will result in the caller being killed without damaging other threads or the kernel.